enterprise/cypher/slotted-runtime/src/main/scala/org/neo4j/cypher/internal/runtime/slotted/expressions/CompiledExpressionConverters.scala

```
@@ -0,0 +1,85 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.cypher.internal.runtime.slotted.expressions
+
+import org.neo4j.cypher.internal.runtime.compiled.expressions.{CodeGeneration, CompiledExpression, IntermediateCodeGeneration}
+import org.neo4j.cypher.internal.runtime.interpreted.ExecutionContext
+import org.neo4j.cypher.internal.runtime.interpreted.commands.convert.{CommunityExpressionConverter, ExpressionConverter, ExpressionConverters}
+import org.neo4j.cypher.internal.runtime.interpreted.commands.expressions.{Expression, RandFunction}
+import org.neo4j.cypher.internal.runtime.interpreted.pipes.QueryState
+import org.neo4j.values.AnyValue
+import org.opencypher.v9_0.expressions.FunctionInvocation
+import org.opencypher.v9_0.expressions.functions.AggregatingFunction
+import org.opencypher.v9_0.{expressions => ast}
+
+object CompiledExpressionConverters extends ExpressionConverter {
+
+  //uses an inner converter to simplify compliance with Expression trait
+  private val inner = new ExpressionConverters(SlottedExpressionConverters, CommunityExpressionConverter)
+
+  override def toCommandExpression(expression: ast.Expression,
+                                   self: ExpressionConverters): Option[Expression] = expression match {
+
+    //we don't deal with aggregations
+    case f: FunctionInvocation if f.function.isInstanceOf[AggregatingFunction] => None
+
+    case e => try {
+      IntermediateCodeGeneration.compile(e) match {
+        case Some(ir) =>
+          Some(CompileWrappingExpression(CodeGeneration.compile(ir),
+                                         inner.toCommandExpression(expression)))
+        case _ => None
+      }
+    } catch {
+      case _: Throwable =>
```

Could we log something instead of just silently eating the exception?